redbot/core/commands/converters.py

```diff
@@ -0,0 +1,74 @@
+import re
+from datetime import timedelta
+from . import Converter, Context, BadArgument
+
+__all__ = ["TimeConverter"]
+
+
+class TimeConverter(Converter):
+    """
+    Convert to a :class:`datetime.timedelta` class.
+
+    The converter supports seconds, minutes, hours and days.
+
+    Time can be attached to the unit (only use the first letter) or not
+    (then use the full word).
+
+    .. admonition:: Example
+
+        *   ``50s`` = ``50 seconds`` -> :class:`datetime.timedelta(seconds=50)`
+        *   ``12m`` = ``12 minutes`` -> :class:`datetime.timedelta(seconds=720)`
+        *   ``4h`` = ``4 hours`` -> :class:`datetime.timedelta(seconds=14400)`
+        *   ``1d`` = ``1 day`` -> :class:`datetime.timedelta(days=1)`
+
+        Using the converter with a command:
+
+        .. code-block:: python3
+
+            @commands.command()
+            async def timer(self, ctx, time: commands.Time):
+                await self.start_timer(time)
+
+        Using the converter manually:
+
+        .. code-block:: python3
+
+            async def convert_time(ctx: Context, text: str) -> datetime.timedelta:
+                time = await commands.Time.convert(ctx, text)
+                return time
+    """
+
+    async def convert(ctx: Context, argument: str) -> timedelta:
+        """
+        Convert manually a string to a :class:`datetime.timedelta` class.
+        
+        .. warning:: This should not be called as a command function annotation.
+            This is for manual calls.
+
+        Arguments
+        ---------
+        ctx: Context
+            The context of the command.
+        argument: str
+            The string you want to convert.
+
+        Returns
+        -------
+        datetime.timedelta
+            The :class:`datetime.timedelta` object.
+
+        Raises
+        ------
+        ~discord.ext.commands.BadArgument
+            No time was found from the given string.
+        """
+        TIME_RE = re.compile(
```

Comment:
This is a compiled constant, it belongs outside the class at the top of the file

Reply:
What do you mean? Put arguments outside of the class and keep what's related to time inside the class?